### DISCUSSION TITLE
为SFC添加过期、废弃、过时的注解

### DISCUSSION THREAD

**Discussion by HighSky2GT** [vuejs/rfcs#515] [General]
### What problem does this feature solve?

类似于使用`@deprecated`标记一个废弃的方法，也可以使用类似的标记使SFC废弃

### What does the proposed API look like?

在SFC头部添加`<!--  @deprecated -->`

**Comment by edison1105** [comment]
感觉应该做到 https://github.com/vuejs/language-tools 里面

**Reply by HighSky2GT** [reply]
是的，应该是需要Volar工具的支持

**Comment by xiaoxiangmoe** [comment]
```vue
<script setup lang="ts">
const props = defineProps<{
    /**
     * 😄
     */
    foobar: string;
}>();
</script>
<script lang="ts">
/**
 * 这是一个组件，
 * 喜欢：
 * * 唱
 * * 跳
 * * rap
 * * 🏀
 *
 * @deprecated 但是过气了
 */
export default {};
</script>
<template>
    <span>🐔</span>
</template>
```

使用效果
<img width="824" alt="image" src="https://github.com/vuejs/core/assets/8111351/42200dfd-5fe4-48ac-967e-a39f2c68bd3d">
<img width="283" alt="image" src="https://github.com/vuejs/core/assets/8111351/52b67077-2d73-4f9b-8812-89a2b74acc57">
<img width="665" alt="image" src="https://github.com/vuejs/core/assets/8111351/093a04e4-ddab-4064-9e6d-095f285ae2f7">

**Reply by ywenhao** [reply]
ji ni tai mei
